src/envs/behavior.py

```diff
@@ -255,36 +255,34 @@ def predicates(self) -> Set[Predicate]:
                 pred = Predicate(pred_name, list(type_combo), classifier)
                 predicates.add(pred)
 
-        # Second, add in custom predicates except reachable-nothing
+        # Second, add in custom predicates except reachable-nothing and
+        # handempty.
         custom_predicate_specs = [
-            ("handempty", self._handempty_classifier, 0),
             ("holding", self._holding_classifier, 1),
-            ("reachable", self._reachable_classifier, 2),
+            ("reachable", self._reachable_classifier, 1),
         ]
 
         for name, classifier, arity in custom_predicate_specs:
             for type_combo in itertools.product(types_lst, repeat=arity):
-                # We only care about reachable when the agent is one of the
-                # types.
                 pred_name = self._create_type_combo_name(name, type_combo)
                 pred = Predicate(pred_name, list(type_combo), classifier)
-                if name == "reachable" and not any(type_i.name == "agent"
-                                                   for type_i in type_combo):
-                    continue
                 predicates.add(pred)
 
-        # Finally, add the reachable-nothing predicate, which only applies
-        # to the 'agent' type
-        for i in range(len(types_lst)):
-            if types_lst[i].name == "agent":
-                pred_name = self._create_type_combo_name(
-                    "reachable-nothing", (types_lst[i], ))
-                pred = Predicate(
-                    pred_name,
-                    [types_lst[i]],
-                    self._reachable_nothing_classifier,
-                )
-                predicates.add(pred)
+        # Finally, add the reachable-nothing and handempty predicates, which
+        # don't take any parameters.
+        pred = Predicate(
+            "reachable-nothing",
+            [],
+            self._reachable_nothing_classifier,
+        )
+        predicates.add(pred)
+
+        pred = Predicate(
```

Comment:
is there a reason that the handempty classifier had to be moved down here? similarly, could reachable-nothing be included in  the `custom_predicate_specs` list  too with 0 arity?

Reply:
good point - that should be correct!